### PR TITLE
Ros2 whisper

### PR DESCRIPTION
**Pull request by maayan25**

### CODE REVIEW

File: common/__init__.py
Comment (jws-1) [inline common/__init__.py:1]: I don't think we need this

File: common/speech/__init__.py
Comment (jws-1) [inline common/speech/__init__.py:1]: I don't think we need this

File: common/speech/lasr_speech_recognition_interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# lasr_speech_recognition_interfaces
+
+Common messages used for speech recognition
+
+This package is maintained by:
+
+- [Maayan Armony](mailto:maayan.armony@gmail.com)
+- [Paul Makles](mailto:me@insrt.uk) (ROS1)
```
Comment (jws-1) [inline common/speech/lasr_speech_recognition_interfaces/README.md:8]:
```suggestion
```

File: common/speech/lasr_speech_recognition_whisper/README.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# lasr_speech_recognition_whisper
+
+Speech recognition implemented using OpenAI Whisper
+
+This package is maintained by:
+
+- [Maayan Armony](mailto:maayan.armony@gmail.com)
+- [Paul Makles](mailto:me@insrt.uk) (ROS1)
```
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/README.md:8]:
```suggestion
```

File: common/speech/lasr_speech_recognition_whisper/__init__.py
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/__init__.py:1]: I don't think this is needed, unless something has changed in ROS2

File: common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/simple_transcribe_microphone.py
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/simple_transcribe_microphone.py:1]: We can probably remove this in favour  of only using `transcribe_microphone_server`

File: common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/transcribe_microphone.py
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/transcribe_microphone.py:1]: Again, we can probably just remove this and keep `transcribe_microphone_server` only

File: common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/transcribe_microphone_server.py [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+    Returns:
+        dict: Dictionary of name: value pairs of command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Starts an action server for transcribing speech."
+    )
+
+    # TODO change to ROS2 rosparams:
+    # port = node.declare_parameter('port', '/dev/ttyUSB0').value
+    # assert isinstance(port, str), 'port parameter must be a str'
```
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/lasr_speech_recognition_whisper/transcribe_microphone_server.py:267]:
```suggestion
```
Let's open up an issue to use rosparam.

File: common/speech/lasr_speech_recognition_whisper/requirements.in
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/requirements.in:1]: Does this work? This is related to the catkin virtualenv (not sure if we found a ROS2 version of that)

File: common/speech/lasr_speech_recognition_whisper/requirements.txt
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/requirements.txt:1]: Same as `requirements.in`

File: common/speech/lasr_speech_recognition_whisper/src/__init__.py
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/src/__init__.py:1]: I think we can remove this whole directory and it's contents, as it's related to the old version of our transcription service

File: common/speech/lasr_speech_recognition_whisper/log/COLCON_IGNORE
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/log/COLCON_IGNORE:1]: Do we need this?

File: common/speech/lasr_speech_recognition_whisper/scripts/__init__.py
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/scripts/__init__.py:1]: Not sure this is needed!

File: common/speech/lasr_speech_recognition_whisper/resource/lasr_speech_recognition_whisper
Comment (jws-1) [inline common/speech/lasr_speech_recognition_whisper/resource/lasr_speech_recognition_whisper:1]: Is this something ROS2 specific & do we need it?

**Review by jws-1 — COMMENTED** [review]
Looks good to me, great work! I haven't been able to test, as I'm having some issues with the ROS2 container on my laptop. Most of the changes requested are to do with deleting stuff that I don't think we necessarily need, but please correct me if I'm wrong!

File: common/speech/lasr_speech_recognition_whisper/requirements.in
Comment (maayan25) [reply common/speech/lasr_speech_recognition_whisper/requirements.in:1]: So far I've used the requirements.in files with pip-tools to find compatible versions across dependencies. We might want to revisit this when more packages are merged, but it should still work with individual virtualenvs that are installed from the def file.

File: common/speech/lasr_speech_recognition_whisper/src/__init__.py
Comment (maayan25) [reply common/speech/lasr_speech_recognition_whisper/src/__init__.py:1]: Done!

File: common/speech/lasr_speech_recognition_whisper/log/COLCON_IGNORE
Comment (maayan25) [reply common/speech/lasr_speech_recognition_whisper/log/COLCON_IGNORE:1]: Probably not, my mistake and removing!

File: common/speech/lasr_speech_recognition_whisper/src/__init__.py
Comment (maayan25) [reply common/speech/lasr_speech_recognition_whisper/src/__init__.py:1]: Actually, need the Cache, so removed everything else

File: common/speech/lasr_speech_recognition_whisper/resource/lasr_speech_recognition_whisper
Comment (maayan25) [reply common/speech/lasr_speech_recognition_whisper/resource/lasr_speech_recognition_whisper:1]: Yes, the package doesn't build without it

**Review by jws-1 — APPROVED** [review]
lgtm, then

**Pull request merged**